src/pages/user/User.tsx

```diff
@@ -78,8 +79,44 @@ const User: React.FC = () => {
   }, [user.authority, dispatch, user.externalOrcid]);
 
   useEffect(() => {
-    setInstitutionPresentations(user.institutionPresentations);
-  }, [user.institutionPresentations, dispatch]);
+    const newInstitutionUnits: InstitutionUnit[] = [];
+    institutionUnits.forEach(institutionUnit => {
+      !user.institutionUnits.find(
+        storedInstitutionUnit => storedInstitutionUnit.cristinUnitId === institutionUnit.cristinUnitId
+      ) ?? newInstitutionUnits.push(institutionUnit);
+    });
+    newInstitutionUnits.forEach(institutionUnit => dispatch(addInstitutionUnit(institutionUnit)));
+  }, [institutionUnits, user.institutionUnits, dispatch]);
+
+  const handleClickAdd = () => setInstitutionUnits([...institutionUnits, emptyInstitutionUnit]);
+
+  const addNewInstitutionUnit = async (cristinUnitId: string) => {
+    const updateRedux = async (cristinUnitId: string) => {
+      if (!user.authority.orgunitids?.find(orgunitid => orgunitid === cristinUnitId)) {
+        const updatedAuthority = await updateInstitutionForAuthority(cristinUnitId, user.authority.systemControlNumber);
+        if (updatedAuthority?.error) {
+          dispatch(addNotification(updatedAuthority.error, 'error'));
+        } else if (updatedAuthority) {
+          dispatch(setAuthorityData(updatedAuthority));
+          try {
+            dispatch(addInstitutionUnit(await institutionLookup(cristinUnitId)));
+          } catch {}
+        }
+      }
+    };
+
+    if (!institutionUnits.find(institutionUnit => institutionUnit.cristinUnitId === cristinUnitId)) {
+      const newInstitutionUnit: InstitutionUnit = await institutionLookup(cristinUnitId);
```

Comment:
hva skjer om `institutionLookup` feiler?

Reply:
samme kall blir gjort i `updateRedux()` som kalles etterpå også, så `newInstitutionUnit` kan sendes videre som argument i stedet, så slipper vi å håndtere eventuelle feil to plasser

Reply:
`institutionLookup` returnerer et objekt med kun cristinUnitId, men er vel ikke en optimal løsning.